extensions/ql-vscode/src/qltest-discovery.ts

```diff
@@ -0,0 +1,206 @@
+import * as path from 'path';
+import { QLPackDiscovery } from './qlpack-discovery';
+import { Discovery } from './discovery';
+import { EventEmitter, Event, Uri, RelativePattern } from 'vscode';
+import { MultiFileSystemWatcher } from 'semmle-vscode-utils';
+import { CodeQLCliServer } from './cli';
+
+/**
+ * A node in the tree of tests. This will be either a `QLTestDirectory` or a `QLTestFile`.
+ */
+export abstract class QLTestNode {
+  constructor(private _path: string, private _name: string) {
+  }
+
+  public get path(): string {
+    return this._path;
+  }
+
+  public get name(): string {
+    return this._name;
+  }
+
+  public abstract get children(): readonly QLTestNode[];
+}
+
+/**
+ * A directory containing one or more QL tests or other test directories.
+ */
+export class QLTestDirectory extends QLTestNode {
+  constructor(_path: string, _name: string, private _children: QLTestNode[]) {
+    super(_path, _name);
+  }
+
+  public get children(): readonly QLTestNode[] {
+    return this._children;
```

Comment:
```suggestion
    return this._children.sort((a, b) => a.path.localeCompare(b.path));
```
Or perhaps there's a better place to do sorting, but we should ensure a sorted order rather than relying on the filesystem to do it for us --- for indeed on linux I see tests in an arbitrary unsorted order.

Reply:
Fixed